src/test/java/com/commercetools/pspadapter/paymentHandler/impl/PaymentHandlerProviderImplTest.java

```diff
@@ -84,15 +93,17 @@ public void whenTenantDoesNotExist_shouldReturnEmptyOptional() {
     @Test
     public void shouldPatchShippingAddress() {
         CtpPaymentWithCart ctpPaymentWithCart = createCartWithPayment();
-        Payment paypalPlusPayment = paymentMapper.ctpPaymentToPaypalPlus(ctpPaymentWithCart);
-        paypalPlusPayment = executeBlocking(this.paypalPlusFacade.getPaymentService().create(paypalPlusPayment));
-
         PaymentHandler paymentHandler = paymentHandlerProvider.getPaymentHandler(MAIN_TEST_TENANT_NAME).get();
-        PaymentHandleResponse paymentHandleResult = paymentHandler.patchAddress(ctpPaymentWithCart.getCart(), paypalPlusPayment.getId());
+        paymentHandler.createPayment(ctpPaymentWithCart.getPayment().getId());
+
+        Cart cartWithExpansion = executeBlocking(this.ctpFacade.getCartService().getByPaymentId(ctpPaymentWithCart.getPayment().getId(),
+                "paymentInfo.payments[*]")).get();
+        String paypalPlusPaymentId = cartWithExpansion.getPaymentInfo().getPayments().get(0).getObj().getInterfaceId();
+        PaymentHandleResponse paymentHandleResult = paymentHandler.patchAddress(cartWithExpansion, paypalPlusPaymentId);
         assertThat(paymentHandleResult.getStatusCode()).isEqualTo(HttpStatus.OK.value());
 
-        paypalPlusPayment = this.paypalPlusFacade.getPaymentService().lookUp(paypalPlusPayment.getId()).toCompletableFuture().join();
-        assertThat(paypalPlusPayment.getTransactions().get(0).getItemList().getShippingAddress()).isNotNull();
+        Payment pPPayment = this.paypalPlusFacade.getPaymentService().lookUp(paypalPlusPaymentId).toCompletableFuture().join();
```

Comment:
do you think maybe there is a better naming for this variable? from the first look on it, doesn't look very straight forward :)

Reply:
@lojzatran seems like this is still unanswered ;)

Reply:
https://github.com/commercetools/commercetools-paypal-plus-integration/pull/49/commits/f123efd870f131e10b3cefcb0b772bece0f38257